x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -1221,18 +1231,25 @@ private void unlink(final LFUCacheEntry entry) {
             assert invariant(entry, false);
         }
 
-        private void maybeEvict() {
+        /**
+         * Cycles through the {@link LFUCacheEntry} from 0 to max frequency and
+         * tries to evict a chunk if no one is holding onto its resources anymore
+         *
+         * @return the frequency of the evicted entry as integer or -1 if no entry was evicted from cache
+         */
+        private int maybeEvict() {
```

Comment:
I think this function should be as follows:
```
        /**
         * Cycles through the {@link LFUCacheEntry} from 0 to max frequency and
         * tries to evict a chunk if no one is holding onto its resources anymore
         *
         * @return the count of the evicted entries.
         */
        private int maybeEvict() {
            assert Thread.holdsLock(SharedBlobCacheService.this);
            for (int currentFreq = 0; currentFreq < maxFreq; currentFreq++) {
                for (LFUCacheEntry entry = freqs[currentFreq]; entry != null; entry = entry.next) {
                    var frequency = entry.freq;
                    boolean evicted = entry.chunk.tryEvict();
                    if (evicted && entry.chunk.io != null) {
                        unlink(entry);
                        keyMapping.remove(entry.chunk.regionKey, entry);
                        return (frequency > 0) ? 1 : 0; 
                    }
                }
            }
            return 0;
        }
```

Reply:
I considered this approach but didn't like it. In your case we are actually using a boolean that tells us if either an entry was evicted with frequency > 0 (return 1) or if 0 is returned there are two cases: either an entry was evicted with frequency 0 or no entry was evicted.
I think the maybeEvict Method should not have the logic for the metic we want and should stay as general as possible.
If someone else uses the maybeEvict it probably would benefit from the frequency returned rather than if the avoided entry had a frequency > 0. WDYT?

Reply:
Fair enough :)